### PR TITLE
UCP: use key only support of Titan for rawkv scan

### LINKED ISSUE #7614: UCP: use key only support of Titan for rawkv scan
## Description

Add key only option for rawKV scan API. It can reduce the message size sending on network and leverage key only feature if Titan enabled. 
- You should set `opts.set_titan_key_only(true)` for raw scan (you can refer to reference https://github.com/tikv/tikv/pull/4966)
- And add an option for client-go https://github.com/tikv/client-go/issues/45
## Difficulty

- Easy

## Score

- 300

## Mentor(s)

- @Connor1996 

## Recommended Skills

- Rust

### PR DESCRIPTION
**Pull request by TommyCpp**
Set key_only to based on the value passed by protobuf call.

Signed-off-by: zhongyang.wu <zhongyang.wu@outlook.com>

### What problem does this PR solve?

Issue Number: close #7614

Problem Summary: Allow Raw KV scan to config Titan to return only keys if applicable

### What is changed and how it works?

The raw KV API already have a `key_only` option so there is no need to change. Also the raw KV API will filter all keys if the `key_only` is set to true.

Only thing change is if the `key_only` in raw Kv API is `true`. Then will set the `key_only` in Titan options to be `true`

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by TommyCpp** [comment]
Blocked on https://github.com/tikv/client-go/pull/47

**Comment by yiwu-arbug** [comment]
I guess this doesn't really block on client-go side of change? If so you can remove the draft state of this PR? Also, please add suitable unit tests for the changes. Thanks for your contribution.

**Comment by TommyCpp** [comment]
> I guess this doesn't really block on client-go side of change? If so you can remove the draft state of this PR? Also, please add suitable unit tests for the changes. Thanks for your contribution.

Hi, I was not sure which kind of test should I do. Discussed with the mentor of the issue, and was recommended to use client-go as a mean of test. See https://github.com/tikv/tikv/issues/7614#issuecomment-620343979

But I got no problem to close this with unit test and I will try to figure out that.

**Comment by yiwu-arbug** [comment]
You can probably duplicate the `test_scan` test in the same file but test whether `key_only` works as expected.

**Comment by yiwu-arbug** [comment]
Please fix CI. You need to run `cargo fmt` once and update the PR.

**Comment by TommyCpp** [comment]
> Looking good overall. One question. The new unit test doesn't seems to enable Titan, how does `storage` able to filter out the values?

Before I implement the `key_only` option in storage, if the incoming request has `key_only` set to true. The storage will just abandon all keys retrived and replace it with empty vector.

So in terms of the functionality of raw scan API, client should feel no change. And to test the logical of this PR, we need to somehow assert the request from API to Titan backend.

File: src/storage/mod.rs [hunk 1]
```diff
@@ -1829,6 +1835,233 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_scan_with_key_only() {
+        let storage = TestStorageBuilder::new().build().unwrap();
```
Comment (Connor1996) [inline src/storage/mod.rs:1840]: how about building it from titan engine

Comment (TommyCpp) [reply src/storage/mod.rs]: Update PR to build from RockDB engine with Titan enbaled.

**Comment by TommyCpp** [comment]
@sre-bot /help

**Comment by TommyCpp** [comment]
@yiwu-arbug @Connor1996 Please help with rerun the integration test.

**Comment by Connor1996** [comment]
@TommyCpp don't forget to update https://github.com/tikv/client-go/pull/47

**Comment by TommyCpp** [comment]
> @TommyCpp don't forget to update [tikv/client-go#47](https://github.com/tikv/client-go/pull/47)

Sorry something came up this weekend. Will try to finish go client tomorrow :smile:

**Pull request merged**